test/common/testlib.py

```diff
@@ -490,14 +490,37 @@ def intercept(test, err):
             if stopTestRun is not None:
                 stopTestRun()
 
-    def setUp(self, macaddr=None, memory_mb=None, cpus=None):
+    # additional_machines is an array of dictionaries, one for each additional machine to be created, e.g.:
+    # additional_machines = [ { machine: { 'image': 'openshift' }, 'start': { 'memory_mb': 1024 } } ]
+    def setUp(self, macaddr=None, memory_mb=None, cpus=None, additional_machines=[]):
```

Comment:
Mutable types as a default argument for a kwarg are usually a bad idea. Better to use None and instantiate to a blank value in the code.